spyder/api/shellconnect/mixins.py

```diff
@@ -1,83 +1,192 @@
-# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# Copyright (c) 2021- Spyder Project Contributors
 #
-# Copyright © Spyder Project Contributors
-# Licensed under the terms of the MIT License
-# (see spyder/__init__.py for details)
+# Released under the terms of the MIT License
+# (see LICENSE.txt in the project root directory for details)
+# -----------------------------------------------------------------------------
 
 """
 Mixins for plugins/widgets that are connected to the IPython console.
 """
 
+from __future__ import annotations
+
+# Standard library imports
+from typing import TYPE_CHECKING
+
+# Third party imports
 from qtpy.QtCore import Signal
 
+# Local imports
 from spyder.api.plugin_registration.decorators import (
-    on_plugin_available, on_plugin_teardown)
+    on_plugin_available,
+    on_plugin_teardown,
+)
 from spyder.api.plugins import Plugins
 
+if TYPE_CHECKING:
+    from qtpy.QtWidgets import QWidget
+
+    import spyder.plugins.ipyconsole.plugin
+    import spyder.plugins.ipyconsole.widgets.shell
```

Comment:
This has the same problem as I mentioned above: `ipyconsole` -> `ipythonconsole`